src/execution_plan/execution_plan.c

```diff
@@ -819,6 +821,19 @@ ExecutionPlan *NewExecutionPlan(RedisModuleCtx *ctx, GraphContext *gc, ResultSet
 	uint *segment_indices = NULL;
 	if(with_clause_count > 0) segment_indices = AST_GetClauseIndices(ast, CYPHER_AST_WITH);
 
+	// We might evaluate an ArithmeticExpression while building an ExecutionPlan, specifically
+	// when attempting to reduce expression trees to scalars.
+	jmp_buf *env = rm_malloc(sizeof(jmp_buf));
+	int res = setjmp(*env);
+	if(res != 0) {
+		// Jumped
+		char *err = QueryCtx_GetError();
+		// The reply arrays have not been instantiated at this point, so we'll return a Redis-level error.
```

Comment:
Maybe we should have the reply array set at this point?

Reply:
If we want, we can build our standard nested reply structure at this point, which might make things easier for clients. This approach looks nicer, though, and I don't conceptually mind the distinction because one case is a compile-time error and one is a run-time error.

Compile-time error:
```
$ gq social "RETURN toUpper(5)"
(error) Type mismatch: expected String but was Integer
```

Run-time error:
```
$ gq social "MATCH (a:person) RETURN toUpper(a.age)"
1) 1) "toUpper(a.age)"
2) (empty list or set)
3) (error) Type mismatch: expected String but was Integer
```

Reply:
This mean that clients will need to check for Redis.Exception while trying to parse the response header.